File: theories/n-2 optimal/n-2.py
def chance(n,b):
    c = 0
    for i in range(1, n+1, 1):
        for j in range(0, n-b, 1):
            c = c + (1/n)*((b/n)**(i-1))*((n-j)/n)**(n-i)
            
    return c

for n in range(4,10000,1):
    biggestValue = 0
    for b in range(n-4,n,1):
        if chance(n,b) > biggestValue:
            biggestValue = chance(n,b)
    if chance(n,n-2) != biggestValue:
        print("logical fallacy")

print("finito")    

File: theories/n-2 optimal/n-2.js

function chance(n,b){
var c=0;

for(i=1;i<n+1;i++){

	for(j=0;j<(n-b);j++){

	c += (1/n)*((b/n)**(i-1))*((n-j)/n)**(n-i);
	
	}

}

return c;

}







for(n=4;n<300;n++){
	
	biggestValue = 0;

	for(b=0; b<n ; b++){

		

		if (chance(n,b) > biggestValue){
			biggestValue = chance(n,b);
		}


	}


	if (chance(n,(n-2)) != biggestValue){

		document.write('logical fallacy at n is'+n+'<br>');
	}


}

console.log('finito');

